### ISSUE TITLE
 Blocks with unicode characters are rendered misaligned

### ISSUE THREAD

**Issue by ooker777** [christianvoigt/argdown#331]
```argdown
[Thesis]: Test
  +  Toán học là nghệ thuật đặt ra cùng một cái tên cho những thứ khác `nhau`
```

![](https://i.imgur.com/voV93Xs.png)

Here is the [sandbox link](https://argdown.org/sandbox/map/?argdown=%5BThesis%5D%3A%20Test%0A%20%20%2B%20To%C3%A1n%20h%E1%BB%8Dc%20l%C3%A0%20ngh%E1%BB%87%20thu%E1%BA%ADt%20%C4%91%E1%BA%B7t%20ra%20c%C3%B9ng%20m%E1%BB%99t%20c%C3%A1i%20t%C3%AAn%20cho%20nh%E1%BB%AFng%20th%E1%BB%A9%20kh%C3%A1c%20nhau)

**Comment by christianvoigt** [comment]
Yes, this is caused by [an issue in the viz.js library](https://github.com/aduh95/viz.js/issues/10) that is hard to debug. This is also causing [#138](https://github.com/christianvoigt/argdown/issues/138) and [#214](https://github.com/christianvoigt/argdown/issues/214).

It would be great if anyone would spend some time digging into the compilation Viz.js to find out what is going wrong there (see link above for further details).

> Yes, although I don't know which libs, and at which step they are required. PR's welcome if anyone is able to get it working.

I am hesitant to apply the "hacky" workaround described there.

**Comment by ooker777** [comment]
Why do you use Viz, when there is Dagre?

**Comment by christianvoigt** [comment]
Good question. The situation regarding rank-based layout algorithms for Javascript is really bad: Viz.js' original developer called it a "hack" and recommended to use Dagre D3 (he subsequently abandoned the project, luckily a fork is still being actively maintained). But Dagre-D3 has not been maintained either for quite a while and is now called "deprecated". Dagre-D3 always lacked many features of Viz.js so it never was a good alternative. But now it looks like it is kind of dead.

yJs offers probably the best layout algorithms, but is a commercial project.

The only good alternative I know of is [Cytoscape](https://js.cytoscape.org/), because they also support the Dagre layout algorithm. I really would like to switch and use the Dagre layout in their framework (instead of using D3). If I have time for a bigger Argdown-related project this would be at the top of my list. For the next months, I will probably not have the time to do this. But if anyone wants to tackle it, I will be glad to help where I can.

**Comment by ooker777** [comment]
I wonder why you didn't use Graphviz at the first place? Is it that you
want a JS lib to be able to embed to website? How about D3-Graphviz or
Vis.js?
